File: python_ruby/class_member.rb
require 'date'
d1 = Date.new(2010, 1, 1)
d2 = Date.new(2020, 2, 29)
p d1.year()
p d2.day()
p Date.today()

class Cs
    def Cs.now() ## class method!
        p "now!"
    end
    def instance_method()
        p "instance~"
    end
end
i = Cs.new()
Cs.now()
i.instance_method()
# Cs.instance_method() error
# i.now() error

class Cs2
    @@count = 0 # 초기화는 맨 처음 한번만
    def initialize()
        @@count += 1
    end
    def Cs.getCount()
        return @@count
    end
end

i1 = Cs2.new()
i2 = Cs2.new()
i3 = Cs2.new()
p Cs.getCount()
# p i1.getCount()

File: python_ruby/class_member.py
class Cs:
    @staticmethod
    def static_method():
        print("static method")
    @classmethod
    def class_method(cls):
        print("class method")
    def instance_method(self):
        print("instance method")

i = Cs()
Cs.static_method()
Cs.class_method()
i.instance_method()

class Cs2:
    count = 0
    def __init__(self):
        Cs2.count += 1
    @classmethod
    def getCount(cls):
        return Cs2.count
i1 = Cs2()
print(i1.count)
print(Cs2.getCount())
i2 = Cs2()
print(i2.count)
i3 = Cs2()
print(Cs2.getCount())
